swagger.yaml

```diff
@@ -1,42 +1,244 @@
 swagger: "2.0"
+
 info:
   title: dp-search-reindex-api
   description: "Provides details about search reindex jobs and enables creation and running of them"
   version: 1.0.0
   license:
     name: "Open Government Licence v3.0"
     url: "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+
 basePath: "/v1"
+
 schemes:
   - http
+
 tags:
-  - name: "hello"
+  - name: "Public"
+
 paths:
-  /hello:
+
+  /jobs/{id}:
     get:
       tags:
-        - hello
-      summary: Example hello world endpoint
-      description: Returns an example payload.
+        - "Public"
+      summary: Get a specific job.
+      description: Get the specific search reindex job that has the id given in the path.
+      parameters:
+        - $ref: '#/parameters/id'
       produces:
         - application/json
       responses:
         200:
           description: OK
           schema:
-            $ref: "#/definitions/HelloResponse"
+            $ref: "#/definitions/Job"
+        500:
+          $ref: '#/responses/InternalError'
+
+  /jobs:
+    get:
+      tags:
+        - "Public"
+      summary: Get a list of jobs.
+      description: Get a list of search reindex jobs that meet the criteria specified by the four query parameters.
+      parameters:
+        - $ref: '#/parameters/limit'
+        - $ref: '#/parameters/offset'
+        - $ref: '#/parameters/sort'
+        - $ref: '#/parameters/state'
+      produces:
+        - application/json
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: "#/definitions/Jobs"
```

Comment:
As it returns a json object containing a list of jobs, Should it be returning the properties such as count,items, limit, offset, total count ,state etc?
An error 400 for invalid request can be added to the response as well

Reply:
Initially the request to get list of jobs actually will not result in any 400 requests (no path, query parameters or a  body to this endpoint) at this point in time.

Future work (stories) will add pagination which could result in 400, but we will add in these details when we come to it. Thanks Sandy :)

Reply:
ok, done

Reply:
After reviewing the rest of the swagger spec, I see we have added query parameters for limit, offset, sort and state so Sandy is right there will be the possibility of 400 bad request.